UefiTestingPkg/AuditTests/PagingAudit/README.md

```diff
@@ -1,5 +1,83 @@
 # Paging Audit
 
+## Contents
+
+- [Paging Audit](#paging-audit)
+  - [Contents](#contents)
+  - [DXE Paging Audit](#dxe-paging-audit)
+    - [DXE Driver](#dxe-driver)
+    - [DXE Shell App](#dxe-shell-app)
+      - [As a Shell-Based Unit Test](#as-a-shell-based-unit-test)
+      - [As a Paging Audit Collection Tool](#as-a-paging-audit-collection-tool)
+  - [SMM Paging Audit](#smm-paging-audit)
+    - [SMM Driver](#smm-driver)
+    - [SMM Shell App](#smm-shell-app)
+  - [Parsing the Paging Data](#parsing-the-paging-data)
+  - [Platform Configuration](#platform-configuration)
+    - [SMM Paging Audit Configuration](#smm-paging-audit-configuration)
+    - [DXE Paging Audit Driver Configuration](#dxe-paging-audit-driver-configuration)
+    - [DXE Paging Audit App Configuration](#dxe-paging-audit-app-configuration)
+  - [Copyright](#copyright)
+
+## DXE Paging Audit
+
+The DXE paging audit is a tool to analyze the page/translation tables of the platform to check
+compliance with Enhanced Memory Protection, debug paging related issues, and understand the
+memory layout of the platform. The DXE version of the audit operates only on the page/translation
+tables of the DXE environment and not the tables used in SMM.
+
+### DXE Driver
+
+The DXE Driver registers an event to be notified on Mu Pre Exit Boot Services (to change this,
+replace gMuEventPreExitBootServicesGuid with a different event GUID), which will then trigger
+the paging information collection. The driver will then save the collected information to
+an available simple file system. The driver version of the DXE paging audit should be used
+when the intent is to capture a snapshot of the page/translation table data at an arbitrary
+point in boot.
+
+### DXE Shell App
+
+The DXE version of UEFI shell app has two modes of operation and **does not require**
+**the DXE driver**.
+
+#### As a Shell-Based Unit Test

```

Comment:
Nit, but I suggest `Mode 1: Shell-Based Unit Test` to clearly associate it as one of two modes mentioned earlier and make it easier to find it as one of two modes in the table of contents.

Reply:
Updated